gems/aws-sdk-core/spec/shared_spec_helper.rb

```diff
@@ -28,6 +28,17 @@
     stub_request(:put, "http://169.254.169.254#{token_path}").to_raise(SocketError)
 
     Aws.shared_config.fresh
+  end
+end
 
+def without_thread_report_on_exception
+  if Thread.respond_to?(:report_on_exception)
+    current_value = Thread.report_on_exception
+    Thread.report_on_exception = false
   end
+
+  yield
```

Comment:
I think a before/after hook is simpler here rather than yielding. I would also think that saving the previous value is a bit overkill.. we don't rely on the exception reporting anyway regardless of Ruby versions. I believe that setting this value only applies for the spec and not for all specs.. but I'm not 100% sure on that one.